flink-streaming-java/src/main/java/org/apache/flink/streaming/api/transformations/StreamTransformation.java

```diff
@@ -221,10 +232,19 @@ public int getMaxParallelism() {
 	 * @param maxParallelism Maximum parallelism for this stream transformation.
 	 */
 	public void setMaxParallelism(int maxParallelism) {
-		Preconditions.checkArgument(maxParallelism > 0
-						&& maxParallelism <= StreamGraphGenerator.UPPER_BOUND_MAX_PARALLELISM,
-				"Maximum parallelism must be between 1 and " + StreamGraphGenerator.UPPER_BOUND_MAX_PARALLELISM
-						+ ". Found: " + maxParallelism);
+		checkArgument(
+				parallelism != ExecutionConfig.PARALLELISM_DEFAULT,
+				"A maximum parallelism can only be specified with an explicitly specified " +
+						"parallelism.");
```

Comment:
This means that we always have to specify a `parallelism` before being able to specify a `maxParallelism`. This seems a bit counter-intuitive to me, as the only constraint seems to be that `parallelism <= maxParalleliem`.